core/sail/shacl/src/test/resources/test-cases/hasValueIn/targetShapeOr/invalid/case2/report.ttl

```diff
@@ -4,3 +4,17 @@
 @prefix sh: <http://www.w3.org/ns/shacl#> .
 @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+
+[] a sh:ValidationReport;
+  sh:conforms false;
+  <http://rdf4j.org/schema/rdf4j#truncated> false;
+  sh:result [ a sh:ValidationResult;
+      sh:focusNode ex:validPerson1;
+      sh:resultPath ex:info2;
+      sh:sourceConstraintComponent sh:MinCountConstraintComponent;
+      sh:resultSeverity sh:Violation;
+      sh:sourceShape [ a sh:PropertyShape;
+          sh:path ex:info2;
+          sh:minCount 1
+        ]
```

Comment:
I haven't seen the actual shape itself being included in a validation report like this before - it's usually just a reference (which is really annoying when the shape has a bnode identifier). Is this new for this reimplementation? And is this what the SHACL spec actually says we _should_ do? If so, I like it a lot :)

Reply:
It's not part of the spec, but due to the way the new AST is implemented we are able to extract a much more minimal shape and add it to the report. Essentially if the shape has both minCount and maxCount but only minCount fails the validation, then maxCount is left out of the shape in the report.